src/ucp/tag/rndv.c

```diff
@@ -455,23 +460,26 @@ static void ucp_rndv_handle_recv_contig(ucp_request_t *rndv_req, ucp_request_t *
             UCS_PROFILE_CALL(uct_rkey_unpack, rndv_rts_hdr + 1,
                              ucp_tag_rndv_rkey_bundle(rndv_req, 0));
         }
-        /* rndv_req is the request that would perform the get operation */
-        rndv_req->send.uct.func     = ucp_proto_progress_rndv_get_zcopy;
-        rndv_req->send.buffer       = rreq->recv.buffer;
-        rndv_req->send.datatype     = ucp_dt_make_contig(1);
-        rndv_req->send.length       = rndv_rts_hdr->size;
-        ucp_request_send_state_reset(rndv_req, ucp_rndv_get_completion,
-                                     UCP_REQUEST_SEND_PROTO_RNDV_GET);
+
         if (rndv_rts_hdr->flags & UCP_RNDV_RTS_FLAG_OFFLOAD) {
-            rndv_req->send.lane     = ucp_ep_get_tag_lane(rndv_req->send.ep);
+            rndv_req->flags    |= UCP_REQUEST_FLAG_OFFLOADED;
+            rndv_req->send.lane = ucp_ep_get_tag_lane(rndv_req->send.ep);
```

Comment:
can we unite code with rndv.c:340?

Reply:
Not sure, here we check RTS packet flags and on line 340 we check request flags